theano/configdefaults.py

```diff
@@ -192,14 +192,6 @@ def filter(val):
     BoolParam(True, allow_override=False),
     in_c_key=False)
 
-
-AddConfigVar('gpuarray.sync',
```

Comment:
You should make it raise an error if used. Keep it. Update the docstring to tell it is removed. Give a lambda as a parameter to the boolparam that raise an error.